File: chroma-league/src/test/java/net/booone/chromaleague/GameStateMocks.java
package net.booone.chromaleague;

import net.booone.chromaleague.rest.activeplayer.ActivePlayer;
import net.booone.chromaleague.rest.activeplayer.ChampionStats;
import net.booone.chromaleague.rest.gamestats.GameStats;
import net.booone.chromaleague.rest.playerlist.Player;
import net.booone.chromaleague.rest.playerlist.PlayerList;
import net.booone.chromaleague.state.RunningState;

import static org.mockito.ArgumentMatchers.eq;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public final class GameStateMocks {
    public static final String PLAYER_RIOT_ID_GAME_NAME = "BonE";
    public static final String PLAYER_RIOT_ID = "BonE#EUPL";

    private final ActivePlayer mockActivePlayer;
    private final PlayerList mockPlayerList;
    private final GameStats mockGameStats;

    public GameStateMocks() {
        RunningState.setRunningGame(false);
        RunningState.setRunningGame(true);
        mockActivePlayer = prepareActivePlayerMock();
        mockPlayerList = preparePlayerListMock();
        mockGameStats = mock(GameStats.class);
        RunningState.getGameState().setPlayerRiotId(PLAYER_RIOT_ID);
    }

    private static PlayerList preparePlayerListMock() {
        PlayerList playerList = mock(PlayerList.class);
        Player activePlayer = mock(Player.class);
        when(playerList.getActivePlayer()).thenReturn(activePlayer);
        return playerList;
    }

    private static ActivePlayer prepareActivePlayerMock() {
        ChampionStats championStats = mock(ChampionStats.class);
        ActivePlayer activePlayer = mock(ActivePlayer.class);
        when(activePlayer.championStats()).thenReturn(championStats);
        when(activePlayer.riotId()).thenReturn(PLAYER_RIOT_ID);
        when(activePlayer.riotIdGameName()).thenReturn(PLAYER_RIOT_ID_GAME_NAME);
        return activePlayer;
    }

    public ChampionStats championStats() {
        RunningState.getGameState().setActivePlayer(mockActivePlayer);
        return mockActivePlayer.championStats();
    }

    public ActivePlayer activePlayer() {
        RunningState.getGameState().setActivePlayer(mockActivePlayer);
        return mockActivePlayer;
    }

    public PlayerList playerList() {
        RunningState.getGameState().setPlayerList(mockPlayerList);
        return mockPlayerList;
    }

    public GameStats gameStats() {
        RunningState.getGameState().setGameStats(mockGameStats);
        return mockGameStats;
    }

    public Player player() {
        RunningState.getGameState().setPlayerList(mockPlayerList);
        return mockPlayerList.getActivePlayer();
    }

    public void mockTestPlayerList() {
        when(playerList().isAlly(eq(PLAYER_RIOT_ID_GAME_NAME))).thenReturn(true);
        when(playerList().isAlly(eq("Ally"))).thenReturn(true);
        when(playerList().isAlly(eq("Enemy"))).thenReturn(false);
    }
}

File: test/GameStateMocks.ts
import {ActivePlayer} from '../src/rest/activeplayer/ActivePlayer.js';
import {ChampionStats} from '../src/rest/activeplayer/ChampionStats.js';
import {GameStats} from '../src/rest/gamestats/GameStats.js';
import {Player} from '../src/rest/playerlist/Player.js';
import {PlayerList} from '../src/rest/playerlist/PlayerList.js';
import {Team} from '../src/rest/playerlist/Team.js';
import {RunningState} from '../src/state/RunningState.js';

export const PLAYER_RIOT_ID_GAME_NAME = 'BonE';
export const PLAYER_RIOT_ID = 'BonE#EUPL';

export class GameStateMocks {
    private mockChampionStats: ChampionStats = {
        attackRange: 0,
        currentHealth: 0,
        maxHealth: 0,
        resourceMax: 0,
        resourceValue: 0,
        resourceType: 'MANA',
    };

    private mockActivePlayer: ActivePlayer = {
        riotId: PLAYER_RIOT_ID,
        riotIdGameName: PLAYER_RIOT_ID_GAME_NAME,
        currentGold: 0,
        level: 1,
        championStats: this.mockChampionStats,
    };

    constructor() {
        RunningState.setRunningGame(false);
        RunningState.setRunningGame(true);
        RunningState.getGameState().playerRiotId = PLAYER_RIOT_ID;
    }

    championStats(): ChampionStats {
        RunningState.getGameState().activePlayer = this.mockActivePlayer;
        return this.mockChampionStats;
    }

    activePlayer(): ActivePlayer {
        RunningState.getGameState().activePlayer = this.mockActivePlayer;
        return this.mockActivePlayer;
    }

    playerList(): PlayerList {
        const player: Player = {
            riotId: PLAYER_RIOT_ID,
            riotIdGameName: PLAYER_RIOT_ID_GAME_NAME,
            team: Team.ORDER,
            championName: 'TestChamp',
            respawnTimer: 0,
            isDead: false,
        };
        const pl = new PlayerList([player]);
        RunningState.getGameState().playerList = pl;
        return pl;
    }

    gameStats(): GameStats {
        const gs: GameStats = { gameMode: 'CLASSIC', gameTime: 0, mapTerrain: 'Default' };
        RunningState.getGameState().gameStats = gs;
        return gs;
    }
}
